src/core/raster/qgscolorrampshader.cpp

```diff
@@ -34,12 +35,26 @@ QgsColorRampShader::QgsColorRampShader( double theMinimumValue, double theMaximu
     , mLUTOffset( 0.0 )
     , mLUTFactor( 1.0 )
     , mLUTInitialized( false )
-    , mColorRampName( QString() )
     , mClip( false )
 {
   QgsDebugMsgLevel( "called.", 4 );
 }
 
+QgsColorRampShader::QgsColorRampShader( const QgsColorRampShader& other ) : QgsRasterShaderFunction( other )
+{
+  mSourceColorRamp.reset( other.sourceColorRamp()->clone() );
```

Comment:
You also need to set all the other member variables here to match the other's values